cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2697,6 +2703,10 @@ if(ARROW_S3)
                             sts)
   endif()
 
+  if(DEFINED ENV{CONDA_PREFIX})
+    unset(BUILD_SHARED_LIBS)
```

Comment:
You're unsetting it even if it was already set.

Reply:
I'm adding a flag for it then.

Reply:
Updated.